src/main/java/com/garbagemule/MobArena/region/ArenaRegion.java

```diff
@@ -524,8 +524,8 @@ private void setPoint(RegionPoint point, Location l) {
         }
 
         // Set the coords and save
-        if (lower != null) setLocation(coords, r1.name().toLowerCase(), lower);
-        if (upper != null) setLocation(coords, r2.name().toLowerCase(), upper);
+        if (lower != null && r1 != null) setLocation(coords, r1.name().toLowerCase(), lower);
+        if (upper != null && r2 != null) setLocation(coords, r2.name().toLowerCase(), upper);
```

Comment:
It took me a while to figure out what the heck the code in this method actually does; it's a bit messy to say the least...

This change is not actually necessary, because in all the branches of the `switch` statement, both `r1` and `r2` are assigned values. Except for the `default` branch, of course, which also assigns `null` to both `upper` and `lower`, which means the "transitive null checks" are sufficient. Also, the `default` branch can't be reached in the current state of the code base, because the only values ever given as RegionPoint are the ones in the four branches.

I can't really tell if it's worth making this code change in order to silence a warning. The underlying issue is that the code is much too messy, I think, so perhaps it would be better with a small refactoring of the method or the class in its entirety?